subsys/bluetooth/audio/has.c

```diff
@@ -70,33 +71,42 @@ static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *at
 }
 
 /* Hearing Access Service GATT Attributes */
-BT_GATT_SERVICE_DEFINE(has_svc,
-	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
-	BT_AUDIO_CHRC(BT_UUID_HAS_HEARING_AID_FEATURES,
-		      BT_GATT_CHRC_READ,
-		      BT_GATT_PERM_READ_ENCRYPT,
-		      read_features, NULL, NULL),
-#if defined(CONFIG_BT_HAS_PRESET_SUPPORT)
-	BT_AUDIO_CHRC(BT_UUID_HAS_PRESET_CONTROL_POINT,
 #if defined(CONFIG_BT_EATT)
-		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE | BT_GATT_CHRC_NOTIFY,
+#define CONTROL_POINT_METHODS (BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE | BT_GATT_CHRC_NOTIFY)
 #else
-		      BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE,
-#endif /* CONFIG_BT_EATT */
-		      BT_GATT_PERM_WRITE_ENCRYPT,
-		      NULL, write_control_point, NULL),
-	BT_AUDIO_CCC(ccc_cfg_changed),
-	BT_AUDIO_CHRC(BT_UUID_HAS_ACTIVE_PRESET_INDEX,
-		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
-		      BT_GATT_PERM_READ_ENCRYPT,
-		      read_active_preset_index, NULL, NULL),
-	BT_AUDIO_CCC(ccc_cfg_changed),
-#endif /* CONFIG_BT_HAS_PRESET_SUPPORT */
-);
+#define CONTROL_POINT_METHODS (BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE)
+#endif
+
+#if defined(CONFIG_BT_HAS_PRESET_SUPPORT)
+#define BT_HAS_PRESET_ATTRS \
+	BT_AUDIO_CHRC(BT_UUID_HAS_PRESET_CONTROL_POINT, \
+		      CONTROL_POINT_METHODS, \
+		      BT_GATT_PERM_WRITE_ENCRYPT, \
+		      NULL, write_control_point, NULL), \
+	BT_AUDIO_CCC(ccc_cfg_changed), \
+	BT_AUDIO_CHRC(BT_UUID_HAS_ACTIVE_PRESET_INDEX, \
+		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY, \
+		      BT_GATT_PERM_READ_ENCRYPT, \
+		      read_active_preset_index, NULL, NULL), \
+	BT_AUDIO_CCC(ccc_cfg_changed)
+#else
+#define BT_HAS_PRESET_ATTRS
+#endif
+
+#define BT_HAS_DEFINITION \
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS), \
+	BT_AUDIO_CHRC(BT_UUID_HAS_HEARING_AID_FEATURES, \
+		      BT_GATT_CHRC_READ, \
+		      BT_GATT_PERM_READ_ENCRYPT, \
+		      read_features, NULL, NULL), \
+	BT_HAS_PRESET_ATTRS
+
+static struct bt_gatt_attr has_attrs[] = { BT_HAS_DEFINITION };
```

Comment:
```suggestion
static struct bt_gatt_attr has_attrs[] = {
	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
	BT_AUDIO_CHRC(BT_UUID_HAS_HEARING_AID_FEATURES,
		      BT_GATT_CHRC_READ,
		      BT_GATT_PERM_READ_ENCRYPT,
		      read_features, NULL, NULL),
#if defined(CONFIG_BT_HAS_PRESET_SUPPORT)
	BT_AUDIO_CHRC(BT_UUID_HAS_PRESET_CONTROL_POINT,
		      CONTROL_POINT_METHODS,
		      BT_GATT_PERM_WRITE_ENCRYPT,
		      NULL, write_control_point, NULL),
	BT_AUDIO_CCC(ccc_cfg_changed),
	BT_AUDIO_CHRC(BT_UUID_HAS_ACTIVE_PRESET_INDEX,
		      BT_GATT_CHRC_READ | BT_GATT_CHRC_NOTIFY,
		      BT_GATT_PERM_READ_ENCRYPT,
		      read_active_preset_index, NULL, NULL),
	BT_AUDIO_CCC(ccc_cfg_changed),
#endif /* CONFIG_BT_HAS_PRESET_SUPPORT */
};
```

Looks cleaner IMO

Reply:
you are right, I'll revert the structure and add it directly in has_attrs